dsl/camel-jbang/camel-jbang-core/src/main/java/org/apache/camel/dsl/jbang/core/commands/infra/InfraRun.java

```diff
@@ -0,0 +1,135 @@
+
+    @CommandLine.Parameters(description = "Service name", arity = "1")
+    private List<String> serviceName;
+
+    public InfraRun(CamelJBangMain main) {
+        super(main);
+    }
+
+    @Override
+    public Integer doCall() throws Exception {
+        if (serviceName == null || serviceName.isEmpty()) {
+            return 0;
+        }
+
+        String service = serviceName.get(0);
+        String serviceImplementation = serviceName.size() > 1 ? serviceName.get(1) : null;
+
+        run(service, serviceImplementation);
+
+        return 0;
+    }
+
+    private void run(String testService, String testServiceImplementation) throws Exception {
+        List<InfraCommand.TestInfraService> metadata;
+
+        try (InputStream is
+                = this.getClass().getClassLoader().getResourceAsStream("META-INF/test-infra-metadata.json")) {
+            String json = new String(is.readAllBytes(), StandardCharsets.UTF_8);
+
+            metadata = InfraCommand.JSON_MAPPER.readValue(json, new TypeReference<List<InfraCommand.TestInfraService>>() {
+            });
+        }
+
+        List<InfraCommand.TestInfraService> services = metadata;
+
+        InfraCommand.TestInfraService testInfraService = services
+                .stream()
+                .filter(service -> {
+                    if (testServiceImplementation != null && !testServiceImplementation.isEmpty()
+                            && service.aliasImplementation() != null) {
+                        return service.alias().contains(testService)
+                                && service.aliasImplementation().contains(testServiceImplementation);
+                    } else if (testServiceImplementation == null) {
+                        return service.alias().contains(testService)
+                                && (service.aliasImplementation() == null || service.aliasImplementation().isEmpty());
+                    }
+
+                    return false;
+                })
+                .findFirst()
+                .orElseThrow(() -> {
+                    String message = ", use the list command for the available services";
+                    if (testServiceImplementation != null) {
+                        return new IllegalArgumentException(
+                                "service " + testService + " with implementation " + testServiceImplementation + " not found"
+                                                            + message);
+                    }
+
+                    return new IllegalArgumentException("service " + testService + " not found" + message);
+                });
+
+        String serviceInterface = testInfraService.service();
+        String serviceImpl = testInfraService.implementation();
+
+        printer().println("Starting service " + testService);
+
+        InfrastructureService actualService = (InfrastructureService) Class.forName(serviceImpl)
+                .getDeclaredConstructor(null)
+                .newInstance(null);
```

Comment:
Food for thought: I think it would be interesting if, in the future, we could pass arguments to the service we are starting, so that it would be possible to configure it in different ways (i.e.: think of something like providing a configuration class for a message broker that configures different protocols, auth mechanisms, behaviors, etc).

Reply:
yeah, that would be nice, so far most of the Services have an empty constructor with a default configuration, but some of them, sftp for example, need a security configuration.

I guess that we can add some information into the annotation, or the Service itself